### PR TITLE
feat: Add volunteer to sign up flow

### PR DESCRIPTION
**Pull request by carissa-tang**
## Implementation description. How did you make this change?
* Added a new step to the signup flow for users to choose `Donor/Volunteer` account type
* Remove `business name` for volunteers
* Make `phone number` not mandatory for volunteers
* Added a mandatory terms and conditions step for volunteers
* Made UI changes to match Figma designs

## Steps to test
1. Go through sign up flow as volunteer - ensure correct fields are shown
2. Go through sign up flow as donor - should be same as before
3. Check mobile and desktop view

https://user-images.githubusercontent.com/69697180/155259057-1a54f035-931c-4191-a0bc-0725cd552e8d.mov
https://user-images.githubusercontent.com/69697180/155259192-e3290214-8393-4a7e-a71b-7cdf936bda1f.mov

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s) 
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] The appropriate tests if necessary have been written

### CODE REVIEW

File: frontend/src/components/auth/Signup/index.tsx [hunk 1]
```diff
@@ -28,13 +31,23 @@ const Signup = () => {
...
+    role: "",
+    acceptedTerms: false,
   });
 
   const { step, navigation }: UseStepType = useStep({ steps, initialStep: 0 });
   const { id } = step;
+  const { next } = navigation;
 
   switch (id) {
+    case "account type":
```
Comment (carissa-tang) [inline frontend/src/components/auth/Signup/index.tsx:43]: not sure if this is the best way to handle "skipping" the terms and conditions page for donors... any suggestions/feedbakc is appreciated!

File: frontend/src/components/auth/Signup/index.tsx [hunk 2]
```diff
@@ -51,6 +64,17 @@ const Signup = () => {
           navigation={navigation}
         />
       );
+    case "terms conditions":
```
Comment (carissa-tang) [inline frontend/src/components/auth/Signup/index.tsx:67]: not sure if this is the best way to handle "skipping" the terms and conditions page for donors... any suggestions/feedback is appreciated!

File: frontend/src/components/common/HeaderLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        </Text>
+      </Box>
+    );
+  }
+  return (
+    <Text mt="67px" textStyle="mobileHeader1">
+      {text}
+    </Text>
+  );
+};
```
Comment (hanlinc27) [inline frontend/src/components/common/HeaderLabel.tsx:23]: seems like you're using `HeaderLabel` in a few places so it's fine to extract like this 
but I don't think we need to pass unnecessary props for it, can do something like:
let me know if it doesn't work?
```suggestion
const HeaderLabel = ({ text }: HeaderLabelProps) =>(
    <Text mt="67px" textStyle="mobileHeader1" textAlign={["inherit", "center"]}>
      {text}
    </Text>
  );
```

File: frontend/src/components/auth/Signup/AccountType.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import { CloseIcon } from "@chakra-ui/icons";
+import {
+  Box,
```
Comment (hanlinc27) [inline frontend/src/components/auth/Signup/AccountType.tsx:3]:
```suggestion
```

File: frontend/src/components/auth/Signup/CreateAccount.tsx [hunk 5]
```diff
@@ -1,21 +1,24 @@
-import { CloseIcon } from "@chakra-ui/icons";
+import { ArrowBackIcon, CloseIcon } from "@chakra-ui/icons";
 import {
   Box,
   Button,
   Container,
   FormControl,
   FormErrorMessage,
+  FormLabel,
```
Comment (hanlinc27) [inline frontend/src/components/auth/Signup/CreateAccount.tsx:8]:
```suggestion
```

File: frontend/src/components/auth/Signup/CreateAccount.tsx [hunk 6]
```diff
@@ -93,70 +101,90 @@ const CreateAccount = ({
...
-          </FormControl>
-        </Box>
-        <Box mt="1rem">
+        <Grid
+          templateColumns={{ md: "repeat(2, 1fr)" }}
+          columnGap={16}
+          rowGap={4}
+        >
+          <Box>
+            <FormControl isRequired isInvalid={!!formErrors.firstName}>
```
Comment (hanlinc27) [inline frontend/src/components/auth/Signup/CreateAccount.tsx:146]: nice job fixing the layout for sign up in general! 😄

File: frontend/src/components/auth/Signup/TermsConditions.tsx [hunk 7]
```diff
@@ -0,0 +1,163 @@
+import { ArrowBackIcon, CloseIcon } from "@chakra-ui/icons";
```
Comment (hanlinc27) [inline frontend/src/components/auth/Signup/TermsConditions.tsx:1]: looking at the figma flow, it seems like the terms and conditions are _after_ the user verifies their email
in which case we wouldn't have all the information in the formData
Can you confirm with design that the flow you have right now is the desired one?

File: frontend/src/components/auth/Signup/index.tsx (hunk 2 above)
Comment (hanlinc27) [reply frontend/src/components/auth/Signup/index.tsx:67]: see comment below

File: frontend/src/components/auth/Signup/index.tsx [hunk 8]
```diff
@@ -51,6 +64,17 @@ const Signup = () => {
...
+      if (formValues.role === Role.VOLUNTEER) {
+        return (
+          <TermsConditions
+            formData={formValues}
+            setForm={setForm}
+            navigation={navigation}
+          />
+        );
+      }
+      return next();
```
Comment (hanlinc27) [inline frontend/src/components/auth/Signup/index.tsx:77]: we shouldn't have the steps be conditional here, rather you should skip it inside of the donor account details, so change it back to 
```suggestion
        return (
          <TermsConditions
            formData={formValues}
            setForm={setForm}
            navigation={navigation}
          />
   )
```

and then in https://github.com/uwblueprint/community-fridge-kw/blob/88498b31174adb59890e9cf2d3e4f9bc5475e160/frontend/src/components/auth/Signup/AccountDetails.tsx#L106
you can do: 
```typescript
    return role===Role.VOLUNTEER ? go("terms conditions") : go("email verification");
```

**Review by hanlinc27 — CHANGES_REQUESTED** [review]

File: frontend/src/components/auth/Signup/TermsConditions.tsx (hunk 7 above)
Comment (carissa-tang) [reply frontend/src/components/auth/Signup/TermsConditions.tsx:1]: confirmed with design that the flow is correct! (the code of conduct comes before the verification email is sent) -- flow is also reflected in figma

**Review by hanlinc27 — APPROVED** [review]

**Pull request merged**